play-frontend-hmrc-play-30/src/main/scala/uk/gov/hmrc/hmrcfrontend/config/TudorCrownConfig.scala

```diff
@@ -0,0 +1,26 @@
+/*
+ * Copyright 2024 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.hmrcfrontend.config
+
+import play.api.Configuration
+
+import javax.inject.Inject
+
+case class TudorCrownConfig @Inject() (config: Configuration) {
```

Comment:
looks fine to me, guess maybe we could make it a @Singleton so there's just one instance unless I've missed anything around that

future thing maybe- noticed that we're doing things a bit different to bootstrap - where they have their case class use simple values then they have a factory to build it from config and in a module they build it to a single instance - I guess for easier unit testing (just do `HmrcBanner(..., TudorCrownConfig(useTudorCrown=false))`) but all our stuff is doing it that way so :shrug:

https://github.com/hmrc/bootstrap-play/blob/05e34001e2b81ee4f3f851caf53e43b5bd1218d0/bootstrap-common-play-30/src/main/scala/uk/gov/hmrc/play/bootstrap/BootstrapModule.scala#L30

Reply:
We cant inject the configuration parameter on an object